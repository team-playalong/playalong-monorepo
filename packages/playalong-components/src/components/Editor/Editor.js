import * as React from 'react';
import PropTypes from 'prop-types';

import {Editor, EditorState} from 'draft-js';
import {stateToHTML} from 'draft-js-export-html';


class PlyEditor extends React.Component {
  constructor(props) {
    super(props);
    this.state = {editorState: EditorState.createEmpty()};
    this.onChange = this.onChange.bind(this);
  }

  onChange(editorState) {
    if (typeof this.props.change === 'function') {
      this.props.change(stateToHTML(editorState.getCurrentContent()));
    }

    this.setState({editorState});
  }

  render() {
    return (
      <Editor editorState={this.state.editorState} onChange={this.onChange} />
    );
  }
}
PlyEditor.propTypes = {
  change: PropTypes.func,
}

export default PlyEditor;
