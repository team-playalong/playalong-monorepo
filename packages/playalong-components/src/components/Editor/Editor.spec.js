import * as React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Editor from './Editor';

describe('Text Input', () => {

  let wrapper;
  const props = {
    label: 'My Text Input',
    value: 'My Val',
  };

  beforeEach(() => {
    wrapper = shallow(<Editor {...props} />);
  });

  test('Snapshot testing', () => {
    const tree = renderer.create(<Editor {...props} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
