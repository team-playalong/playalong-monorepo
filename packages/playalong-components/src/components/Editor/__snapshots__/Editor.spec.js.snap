// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Input Snapshot testing 1`] = `
<div
  className="DraftEditor-root"
>
  <div
    className="DraftEditor-editorContainer"
  >
    <div
      aria-activedescendant={undefined}
      aria-autocomplete={undefined}
      aria-describedby={null}
      aria-expanded={undefined}
      aria-haspopup={undefined}
      aria-label={undefined}
      aria-owns={undefined}
      autoCapitalize={undefined}
      autoComplete={undefined}
      autoCorrect={undefined}
      className="notranslate public-DraftEditor-content"
      contentEditable={true}
      data-testid={undefined}
      onBeforeInput={[Function]}
      onBlur={[Function]}
      onCompositionEnd={[Function]}
      onCompositionStart={[Function]}
      onCopy={[Function]}
      onCut={[Function]}
      onDragEnd={[Function]}
      onDragEnter={[Function]}
      onDragLeave={[Function]}
      onDragOver={[Function]}
      onDragStart={[Function]}
      onDrop={[Function]}
      onFocus={[Function]}
      onInput={[Function]}
      onKeyDown={[Function]}
      onKeyPress={[Function]}
      onKeyUp={[Function]}
      onMouseUp={[Function]}
      onPaste={[Function]}
      onSelect={[Function]}
      role="textbox"
      spellCheck={false}
      style={
        Object {
          "outline": "none",
          "whiteSpace": "pre-wrap",
          "wordWrap": "break-word",
        }
      }
      suppressContentEditableWarning={true}
      tabIndex={undefined}
    >
      <div
        data-contents="true"
      >
        <div
          className=""
          data-block={true}
          data-editor="d0820"
          data-offset-key="cd3po-0-0"
        >
          <div
            className="public-DraftStyleDefault-block public-DraftStyleDefault-ltr"
            data-offset-key="cd3po-0-0"
          >
            <span
              data-offset-key="cd3po-0-0"
              style={Object {}}
            >
              <br
                data-text="true"
              />
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
